src/main/drivers/bus_spi_hal.c

```diff
@@ -295,10 +295,10 @@ static bool spiBusReadBuffer(const busDevice_t *bus, uint8_t *out, int len)
 }
 
 // return uint8_t value or -1 when failure
-uint8_t spiTransferByte(SPI_TypeDef *instance, uint8_t in)
+uint8_t spiTransferByte(SPI_TypeDef *instance, uint8_t data)
 {
-    spiTransfer(instance, &in, &in, 1);
-    return in;
+    spiTransfer(instance, &data, &data, 1);
```

Comment:
It would be better to allocate local buffer here - there is no guarantee that  spiTransfer can use one buffer, it may prevent future optimizations